### PR TITLE
Upgrade next-i18next package

### PR DESCRIPTION
**Pull request by aaronmgdr**
Welcome to The Methuselah of package upgrades. 

going from version 6 to 8 requires big changes as we are no longer using `getInitialProps` instead each page with translations must defined a `getServerSideProps` or `getStaticProps` function. which yes means that for the first time some of our pages will be 'static'. (praise be). 

Bonus here is that the shared bundle sized decreased.

Bonus Bonus is that I fixed faucet on dev. (a secret was missing)

### CODE REVIEW

File: src/_page-tests/test-utils.tsx [hunk 1]
```diff
@@ -1,17 +1,16 @@
 import { render } from "@testing-library/react"
 import * as React from "react"
-import NextI18NextInstance from "src/utils/i18nForTests"
-
+import i18nForTests from "src/utils/i18nForTests"
+// import i18nConfig from "src/../next-i18next.config.js"
```
Comment (carterqw2) [inline src/_page-tests/test-utils.tsx:4]: Could be deleted perhaps?

File: src/dev/ValidatorsListApp.tsx [hunk 2]
```diff
@@ -25,11 +24,15 @@ const networkMenu = [
   // ['Baklavastaging', menuItems.VALIDATORS_LIST_BAKLAVASTAGING.link],
```
Comment (carterqw2) [inline src/dev/ValidatorsListApp.tsx:24]: Maybe we could clean it up as well?

File: src/dev/ValidatorsListApp.tsx [hunk 3]
```diff
@@ -25,11 +24,15 @@ const networkMenu = [
...
 ]
 
+const BLOCKSCOUT = {
+  baklava: "https://baklava-blockscout.celo-testnet.org/graphiql",
+  baklavastaging: "https://baklavastaging-blockscout.celo-testnet.org/graphiql",
```
Comment (carterqw2) [inline src/dev/ValidatorsListApp.tsx:29]: I'm not sure it even exists anymore.

File: src/i18n.ts [hunk 4]
```diff
@@ -1,27 +1,21 @@
...
-export const Trans = NextI18NextInstance.Trans
+export {
+  Trans,
+  appWithTranslation,
+  withTranslation,
+  useTranslation,
+  withTranslation as withNamespaces,
+} from "next-i18next"
+
+// export const Trans = NTrans
```
Comment (carterqw2) [inline src/i18n.ts:11]: Leftover?

File: src/utils/i18nForTests.ts [hunk 5]
```diff
@@ -1,26 +1,25 @@
...
-      [NameSpaces.community]: require("public/static/locales/en/community.json"),
-      [NameSpaces.common]: require("public/static/locales/en/common.json"),
-      [NameSpaces.dev]: require("public/static/locales/en/dev.json"),
-      [NameSpaces.download]: require("public/static/locales/en/download.json"),
-      [NameSpaces.faucet]: require("public/static/locales/en/faucet.json"),
-      [NameSpaces.home]: require("public/static/locales/en/home.json"),
-      [NameSpaces.jobs]: require("public/static/locales/en/jobs.json"),
-      [NameSpaces.terms]: require("public/static/locales/en/terms.json"),
-      [NameSpaces.papers]: require("public/static/locales/en/papers.json"),
+      [NameSpaces.about]: require("public/locales/en/about.json"),
```
Comment (carterqw2) [inline src/utils/i18nForTests.ts:9]: What about other namespaces, like `cbe` etc, shouldn't they be listed here as well?

File: pages/code-of-conduct.tsx [hunk 6]
```diff
@@ -1,3 +1,15 @@
 import CodeOfConduct from "src/community/CodeOfConduct"
 
+import { serverSideTranslations } from "next-i18next/serverSideTranslations"
+import { NameSpaces } from "src/i18n"
+
+export async function getStaticProps() {
+  return {
+    props: {
+      ...(await serverSideTranslations("en", [NameSpaces.common, NameSpaces.cbe])),
```
[the remark this answers is not in the document]
Comment (aaronmgdr) [reply pages/code-of-conduct.tsx:9]: great catch!

File: src/_page-tests/test-utils.tsx (hunk 1 above)
Comment (aaronmgdr) [reply src/_page-tests/test-utils.tsx:4]: yes

File: src/dev/ValidatorsListApp.tsx (hunk 2 above)
Comment (aaronmgdr) [reply src/dev/ValidatorsListApp.tsx:24]: yes

**Review by carterqw2 — APPROVED** [review]
Looks good but the cbe test is failing.

**Pull request merged**